### PR TITLE
add support for mayavi

### PR DESCRIPTION
**Pull request by agramfort**

### CODE REVIEW

**Comment by agramfort** [comment]
I fear a rebase mess here. No time to do it.

please take over.

File: sphinxgallery/gen_rst.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import posixpath
 import subprocess
 import warnings
+from copy import deepcopy
```
Comment (Titan-C) [inline sphinxgallery/gen_rst.py]: this deepcopy is not used?

File: sphinxgallery/gen_rst.py [hunk 2]
```diff
@@ -987,6 +1040,9 @@ def embed_code_links(app, exception):
         'numpy': 'http://docs.scipy.org/doc/numpy-1.6.0',
         'scipy': 'http://docs.scipy.org/doc/scipy-0.11.0/reference',
     }
+    if has_mayavi:
```
Comment (Titan-C) [inline sphinxgallery/gen_rst.py]: The new version of sphinx-gallery handles this urls as user input. Shall this considered as default documented package?

File: sphinxgallery/gen_rst.py (hunk 1 above)
Comment (agramfort) [inline sphinxgallery/gen_rst.py]: yes please remove it.

File: sphinxgallery/gen_rst.py (hunk 2 above)
Comment (agramfort) [inline sphinxgallery/gen_rst.py]: not a default, just optional. I'll need it for mne-python.

File: sphinxgallery/gen_rst.py [hunk 3]
```diff
@@ -118,6 +129,31 @@ def _get_data(url):
...
+    """Scale image such that width <= max_width
+    """
+    try:
+        from PIL import Image
+    except ImportError:
+        import Image
+    img = Image.open(in_fname)
+    width_in, height_in = img.size
+
+    if width_in <= max_width:
```
Comment (Titan-C) [inline sphinxgallery/gen_rst.py]: this is possibly the only extra clause that distinguishes this function, from the function that does the thumbnails.
Is there a reason to use this instead than the `make_thumbnail` with different input parameters and add there a clause like 
```
if scale >1.0:
    return
```

**Comment by agramfort** [comment]
closing in favor or #17

**Pull request closed without merging**